File: JavaScript/1975-Maximum Matrix Sum.js
/**
 * @param {number[][]} matrix
 * @return {number}
 */
var maxMatrixSum = function(matrix) {
    let row = matrix.length, col = matrix[0].length;
    let res = 0, bad = 0, absMin = Number.POSITIVE_INFINITY;
    
    for (let r=0; r<row; r++){
        for (let c=0; c<col; c++){
            let absNum = Math.abs(matrix[r][c]);
            res += absNum;
            if (matrix[r][c] < 0) bad++;
            absMin = Math.min(absMin, absNum);
        }
    }
    
    return bad%2 ? res-2*absMin : res;
};

File: C++/1975-Maximum Matrix Sum.cpp
using LL = long long;
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        LL tot = 0;
        int row = matrix.size(), col = matrix[0].size(), minusCnt = 0, mn = INT_MAX;
        for (int r=0; r<row; r++) {
            for (int c=0; c<col; c++) {
                tot += abs(matrix[r][c]);
                mn = min(mn, abs(matrix[r][c]));
                if (matrix[r][c] <= 0) {
                    minusCnt++;
                }
            }
        }

        if (minusCnt%2 == 0) {
            return tot;
        } else {
            return tot - 2*mn;
        }

    }
};